### ISSUE TITLE
Ignore GStringExpressionWithinString when inside of sh step (Jenkinsfile)

### ISSUE THREAD

**Issue by felipecrs** [nvuillam/npm-groovy-lint#143]
I'm not sure if this is doable or not, but it would be great if this rule could be automatically ignored when checking for GStrings expressions within `sh` steps in Jenkinsfiles, since they actually mean environment variables accessible by the shell interpreter.

E.g.:

```groovy
sh '''
echo $SHELL
'''
```

**Comment by nvuillam** [comment]
That seems  to be a false positive... maybe we could ignore it just for jenkinsfile ?

**Comment by felipecrs** [comment]
> That seems to be a false positive... maybe we could ignore it just for jenkinsfile ?

This is what I do currently, but it would be more assertive if we could ignore it only for the `sh` steps.
